x-pack/legacy/plugins/infra/public/components/source_configuration/log_columns_configuration_panel.tsx

```diff
@@ -30,70 +35,95 @@ interface LogColumnsConfigurationPanelProps {
   isLoading: boolean;
   logColumnConfiguration: LogColumnConfigurationProps[];
   addLogColumn: (logColumn: LogColumnConfiguration) => void;
+  moveLogColumn: (sourceIndex: number, destinationIndex: number) => void;
 }
 
 export const LogColumnsConfigurationPanel: React.FunctionComponent<
   LogColumnsConfigurationPanelProps
-> = ({ addLogColumn, availableFields, isLoading, logColumnConfiguration }) => (
-  <EuiForm>
-    <EuiFlexGroup>
-      <EuiFlexItem>
-        <EuiTitle size="s" data-test-subj="sourceConfigurationLogColumnsSectionTitle">
-          <h3>
-            <FormattedMessage
-              id="xpack.infra.sourceConfiguration.logColumnsSectionTitle"
-              defaultMessage="Columns"
-            />
-          </h3>
-        </EuiTitle>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>
-        <AddLogColumnButtonAndPopover
-          addLogColumn={addLogColumn}
-          availableFields={availableFields}
-          isDisabled={isLoading}
-        />
-      </EuiFlexItem>
-    </EuiFlexGroup>
-    {logColumnConfiguration.length > 0 ? (
-      logColumnConfiguration.map((column, index) => (
-        <LogColumnConfigurationPanel
-          logColumnConfigurationProps={column}
-          key={`logColumnConfigurationPanel-${index}`}
-        />
-      ))
-    ) : (
-      <LogColumnConfigurationEmptyPrompt />
-    )}
-  </EuiForm>
-);
+> = ({ addLogColumn, moveLogColumn, availableFields, isLoading, logColumnConfiguration }) => {
+  const onDragEnd = useCallback(
+    ({ source, destination }: DropResult) =>
+      destination && moveLogColumn(source.index, destination.index),
+    []
```

Comment:
Shouldn't `moveLogColumn` be part of the dependencies array? Its identity changes on every column state change.

```suggestion
    [moveLogColumn]
```